### PR TITLE
HttpClient Metrics instrumentation for .NET8

### LINKED ISSUE #3054: Support .NET 8 - HttpClient metrics
Follow up to #2996 and #3051.

https://github.com/open-telemetry/opentelemetry-dotnet/commit/d07d03086ae269174a94552448cfd0bcf11aa37a

### PR DESCRIPTION
**Pull request by Kielek**
## Why

Fixes #3054

## What

Apply changes from https://github.com/open-telemetry/opentelemetry-dotnet/commit/d07d03086ae269174a94552448cfd0bcf11aa37a

Changelog already covered by:
- `OpenTelemetry.Instrumentation.GrpcNetClient`,
  `OpenTelemetry.Instrumentation.Http`,
  and `OpenTelemetry.Instrumentation.SqlClient` updated from `1.5.1-beta.1` to `1.6.0-beta.2`.

## Tests

CI + modify existing tests

## Checklist

- ~~[ ] `CHANGELOG.md` is updated.~~
- ~~[ ] Documentation is updated.~~
- [x] New features are covered by tests.

### CODE REVIEW

File: src/OpenTelemetry.AutoInstrumentation/Util/DotNetVersionHelper.cs
Comment (Kielek) [inline src/OpenTelemetry.AutoInstrumentation/Util/DotNetVersionHelper.cs:22]: Similar to what is done in 
```
namespace System.Runtime.InteropServices
{
    public static partial class RuntimeInformation
    {
        private const string FrameworkName = ".NET";
        private static string? s_frameworkDescription;
        private static string? s_runtimeIdentifier;

        public static string FrameworkDescription
        {
            get
            {
                if (s_frameworkDescription == null)
                {
                    ReadOnlySpan<char> versionString = typeof(object).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion;

                    // Strip the git hash if there is one
                    int plusIndex = versionString.IndexOf('+');
                    if (plusIndex >= 0)
                    {
                        versionString = versionString.Slice(0, plusIndex);
                    }

                    s_frameworkDescription = !versionString.Trim().IsEmpty ? $"{FrameworkName} {versionString}" : FrameworkName;
                }

                return s_frameworkDescription;
            }
        }
```

File: test/IntegrationTests/HttpTests.cs
Comment (Kielek) [inline test/IntegrationTests/HttpTests.cs:89]: Ensure that we do not fetched `OpenTelemetry.Instrumentation.Http`. It should be not enabled for .NET8 as in Instrumentation.Http package.

File: src/OpenTelemetry.AutoInstrumentation/Configurations/EnvironmentConfigurationMetricHelper.cs
Comment (pjanotti) [inline src/OpenTelemetry.AutoInstrumentation/Configurations/EnvironmentConfigurationMetricHelper.cs:111]:
```suggestion
                return builder
                    .AddMeter("System.Net.Http")
                    .AddMeter("System.Net.NameResolution");
```

**Review by pjanotti — APPROVED** [review]

File: src/OpenTelemetry.AutoInstrumentation/Configurations/EnvironmentConfigurationMetricHelper.cs
Comment (rajkumar-rangaraj) [inline src/OpenTelemetry.AutoInstrumentation/Configurations/EnvironmentConfigurationMetricHelper.cs:107]: This is little confusing as the project has only .NET6.0. Leaving a comment would help.

**Review by rajkumar-rangaraj — APPROVED** [review]

File: src/OpenTelemetry.AutoInstrumentation/Configurations/EnvironmentConfigurationMetricHelper.cs
Comment (Kielek) [reply src/OpenTelemetry.AutoInstrumentation/Configurations/EnvironmentConfigurationMetricHelper.cs:107]: Project is targeted also for .NET Framework 4.6.2.

I have added comment in f7eeae793ffcf06bb360815057fccb62b67885da. I am gonna merge it to unblock further work on .NET8 release. Please let me know, even after merge, if it is more clear now.

File: src/OpenTelemetry.AutoInstrumentation/Configurations/EnvironmentConfigurationMetricHelper.cs
Comment (Kielek) [reply src/OpenTelemetry.AutoInstrumentation/Configurations/EnvironmentConfigurationMetricHelper.cs]: Part of 7eeae793ffcf06bb360815057fccb62b67885da

File: src/OpenTelemetry.AutoInstrumentation/Configurations/EnvironmentConfigurationMetricHelper.cs
Comment (rajkumar-rangaraj) [reply src/OpenTelemetry.AutoInstrumentation/Configurations/EnvironmentConfigurationMetricHelper.cs:107]: I understand it. When someone else reviews `NET6_0_OR_GREATER` maybe misleading as it got only .NET 6.0, part from net462. In reality we are compiling only NET6 bits here. This is not a blocker, if you want to merge please proceed.

File: src/OpenTelemetry.AutoInstrumentation/Util/DotNetVersionHelper.cs
Comment (nrcventura) [reply src/OpenTelemetry.AutoInstrumentation/Util/DotNetVersionHelper.cs:22]: Would Environment.Version be a better choice here? For .net 6 and higher it should be reporting the correct version.

**Pull request merged**
